specification/network/resource-manager/Microsoft.Network/stable/2025-05-01/virtualNetworkAppliance.json

```diff
@@ -0,0 +1,465 @@
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "A virtual network appliance in a resource group."
+    },
+    "VirtualNetworkApplianceIpConfiguration": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualNetworkApplianceIpConfigurationProperties",
+          "description": "Properties of the virtual network appliance ip configuration."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of virtual network appliance ip configuration."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "The virtual network appliance ip configuration."
+    },
+    "VirtualNetworkApplianceIpConfigurationProperties": {
+      "type": "object",
+      "properties": {
+        "privateIPAddress": {
+          "type": "string",
+          "description": "The private IP address of the IP configuration."
+        },
+        "privateIPAllocationMethod": {
+          "$ref": "./network.json#/definitions/IPAllocationMethod",
+          "description": "The private IP address allocation method."
+        },
+        "primary": {
+          "type": "boolean",
+          "description": "Whether the ip configuration is primary or not."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the private link service IP configuration resource."
+        },
+        "privateIPAddressVersion": {
+          "$ref": "./network.json#/definitions/IPVersion",
+          "description": "Whether the specific IP configuration is IPv4 or IPv6. Default is IPv4."
+        }
+      },
+      "description": "Properties of virtual network appliance IP configuration."
+    },
+    "VirtualNetworkApplianceListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VirtualNetworkAppliance"
+          },
+          "description": "A list of virtual network appliances in a resource group."
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URL to get the next set of results."
+        }
+      },
+      "description": "Response for the ListVirtualNetworkAppliance API service call."
+    }
+  },
+  "parameters": {
+    "VirtualNetworkApplianceNameParameter": {
```

Comment:
This should not be a client level parameter. You need to add `"x-ms-parameter-location": "method"`. Refer [this](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-parameter-location).